### DISCUSSION TITLE
`use` on key/mouse events

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#148] [General]
I suppose it is not possible to use use a `WidgetHandle` on these kind of async events. So I can only think on using `CapsuleContainer.read`.

Is that right?

UPDATE: I can think too on save a controller on a class property.

**Comment by GregoryConrad** [comment]
If you mean:

```dart
MyWidget(
  onPressed: () => use(something).thing(),
)
```

Ideally don’t do that. Instead, do:

```dart
final thing = use(something).thing;
MyWidget(
  onPressed: thing,
)
```

**Reply by busslina** [reply]
Sometimes I just make myself a mess between OOP and FP
